wsmaster/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineManager.java

```diff
@@ -1037,6 +1023,37 @@ private void cleanup() {
             executor.shutdownNow();
         }
 
+        final ExecutorService destroyMachinesExecutor =
+                Executors.newFixedThreadPool(10 * Runtime.getRuntime().availableProcessors(),
```

Comment:
I think that `10 * Runtime.getRuntime().availableProcessors()` is too many :open_mouth: :gun: , i think that ` 2 * Runtime.getRuntime().availableProcessors()` or even  `Runtime.getRuntime().availableProcessors()` will be more than enough.